spec/std/humanize_spec.cr

```diff
@@ -29,6 +29,16 @@ describe Number do
       123_456.format(delimiter: '.').should eq "123.456"
 
       123_456.789.format.should eq "123,456.789"
+
+      1e15.format(decimal_places: 7).should eq "1,000,000,000,000,000.0000000"
+      1e15.to_i64.format(decimal_places: 7).should eq "1,000,000,000,000,000.0000000"
+      1e-5.format(decimal_places: 7).should eq "0.0000100"
+      1e-4.format(decimal_places: 7).should eq "0.0001000"
+
+      Float64::MAX.format.should eq "179,769,313,486,231,570,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000.0"
```

Comment:
In Ruby p "%f" % Float::MAX prints something different (179769313486231570814527423731704356798070567525844996598917476803157260780028538760589558632766878171540458953514382464234321326889464182768467546703537516986049910576551282076245490090389328944075868508455133942304583236903222948165808559332123348274797826204144723168738177180919299881250404026184124858368.000000). I wonder where the difference is coming from.

Reply:
I suppose our float printing algorithm probably just doesn't have enough precision for this.
@HertzDevil Any thoughts?

Reply:
`%f` in C and Ruby produces exact digits before the decimal place, since `Float::Printer` produces the shortest round-trip it is unsuitable for `%f`. On Crystal master `"%f" % Float64::MAX` should also produce that exact number (right now this comes from `LibC.snprintf`).

Reply:
Isn't it the case that Ruby has taken a couple of bits for the type marker (google for `ruby immediate values` if you want a more in depth explanation), so Float in Ruby is not Float64 but something like Float62 or something like that.